### ISSUE TITLE
C Parser fails to parse typedef inside a function

### ISSUE THREAD

**Issue by xiaoyinl** [NationalSecurityAgency/ghidra#3526]
**Describe the bug**
Ghidra can't parse the following C code:
```
void testFunc()
{
    typedef int MYTESTFN(int);
    MYTESTFN* func = (MYTESTFN*)0;
}
```
The error is:
```
Parse Errors: C Parser:  Encountered errors during parse.
         in /home/.../test/test.c near line 2
         near token: )
         Last Valid Datatype: testFunc
         Check around CParserPlugin.out around line: 6
```
CParserPlugin.out is:
```

#line 1: "/home/.../test/test.c"
void testFunc()
{
	typedef int MYTESTFN(int);
	MYTESTFN* func = (MYTESTFN*)0;
}

```
(Line 6 is `MYTESTFN* func = (MYTESTFN*)0;`)

**To Reproduce**
1. Save the above C code as test.c
1. Open CodeBrowser
2. Open File -> Parse C Source...
3. Add `test.c`. Then click "Parse to File..." Save as "test.gdt".
4. Parse Errors occurs

**Expected behavior**
Ghidra should successfully parse it.

**Environment (please complete the following information):**
 - OS: Kali Linux x64
 - Java Version: OpenJDK 11.0.12
 - Ghidra Version: master (f9463e600d318a74e81462656e4a87f3e598ddec)
 - Ghidra Origin: locally built

**Additional context**
I found this issue when I tried to parse the Windows driver header ntddk.h. A parse error occurs in this function:
```
__forceinline NTSTATUS RtlxQueryRegistryValues(ULONG RelativeTo, PCWSTR Path, PRTL_QUERY_REGISTRY_TABLE QueryTable, PVOID Context, PVOID Environment)
{
    typedef NTSTATUS __stdcall QUERYFN(ULONG, PCWSTR, PRTL_QUERY_REGISTRY_TABLE, PVOID, PVOID);
    UNICODE_STRING FunctionName;
    QUERYFN* QueryRoutine;
    RtlInitUnicodeString(&FunctionName, L"RtlQueryRegistryValuesEx");
    QueryRoutine = (QUERYFN*)MmGetSystemRoutineAddress(&FunctionName);
    if (QueryRoutine == ((void *)0)) {
        QueryRoutine = &RtlQueryRegistryValues;
    }
    return QueryRoutine(RelativeTo, Path, QueryTable, Context, Environment);
}
```

**Comment by xiaoyinl** [comment]
@ryanmkurtz @emteere Since you are working on improving the C Parser, could you please consider fixing this one as well? This one and #4660 are two remaining issues that cause errors in parsing the preprocessed Windows WDK headers. Thank you!

**Comment by xiaoyinl** [comment]
Fixed by ccee80e8ad9853f90891ae2d9ace1e304ba71a24
